index.js

```diff
@@ -101,44 +101,34 @@ const getAddresses = () => {
 
 
 const hash = (address) => {
-  return Crypto.SHA256(address).toString(Crypto.enc.Hex);
-
+  return Crypto.SHA256(address).toString(Crypto.enc.Hex)
 }
 
 const setTotalActiveUsers = (req) => {
   const address = hash(req.connection.remoteAddress)
-  incrementActiveUsers(address)
-}
-
-incrementActiveUsers = (address) => {
   if(!userActive(address)){ //The last connection was before we zeroed the totalActiveUsers field
     _totalActiveUsers++
   }
 }
 
-decrementActiveUsers = (address) => {
-  if (userActive(address)) { //The last connection was after we zeroed the totalActiveUsers field
-    _totalActiveUsers--
-  }
-}
-
 const userActive = (address) => {  
-  const lastRequestAt = _rlAddressToRequests[address] ? _rlAddressToRequests[address].lastRequestAt : null
+  const lastRequestAt = _rlAddressToRequests[address] && _rlAddressToRequests[address].lastRequestAt
   return (lastRequestAt && lastRequestAt.diff(_lastActiveTimeout) >= 0)
 }
 
 const removeUser = (address) => {
-  decrementActiveUsers(address) //If user is active we must decrement the active users field
+  if (userActive(address)) { //The last connection was after we zeroed the totalActiveUsers field
+    _totalActiveUsers--
+  }
   delete _rlAddressToRequests[address]
-  
 }
 
 const addUser = (address, userTableMaxSize, now) => {
-  _userQueue.push(address)
   if (Object.keys(_rlAddressToRequests).length >= userTableMaxSize) { //We have reached an overflow scenario
-    let addressToRemove = _userQueue.shift()
+    const addressToRemove = _userQueue.shift()
```

Comment:
Now we're only removing one user when the list is full right? Is that optimal or should we maybe remove ~10 %? Don't know if that's better performance-wise but I remember that we discussed it

Reply:
Well if we remove 10% the application don't scale that well. If the user table are large enough removing 10% of them could impose a heavy toll on  the single user which triggers the overflow. I don't think that the overhead will be smaller if we do it in batches?

Reply:
Yes makes sense, let's keep it this way